sdk/spring/spring-cloud-azure-feature-management/src/main/java/com/azure/spring/cloud/feature/management/implementation/timewindow/recurrence/RecurrenceEvaluator.java

```diff
@@ -0,0 +1,158 @@
+        }
+
+        return occurrenceInfo;
+    }
+
+    /**
+     * Find the closest previous recurrence occurrence before the provided time stamp according to the "Daily" recurrence pattern.
+     *
+     * @return The return result contains two property, one is previousOccurrence, the other is numberOfOccurrences.
+     * previousOccurrence: The closest previous occurrence.
+     * numberOfOccurrences: The number of complete recurrence intervals which have occurred between the time and the recurrence start.
+     */
+    private static OccurrenceInfo getDailyPreviousOccurrence(TimeWindowFilterSettings settings, ZonedDateTime now) {
+        final ZonedDateTime start = settings.getStart();
+        final int interval = settings.getRecurrence().getPattern().getInterval();
+        final int numberOfOccurrences = (int) (Duration.between(start, now).getSeconds() / Duration.ofDays(interval).getSeconds());
+        return new OccurrenceInfo(start.plusDays((long) numberOfOccurrences * interval), numberOfOccurrences + 1);
+    }
+
+    /**
+     * Find the closest previous recurrence occurrence before the provided time stamp according to the "Weekly" recurrence pattern.
+     *
+     * @return The return result contains two property, one is previousOccurrence, the other is numberOfOccurrences.
+     * previousOccurrence: The closest previous occurrence.
+     * numberOfOccurrences: The number of recurring days of week which have occurred between the time and the recurrence start.
+     */
+    private static OccurrenceInfo getWeeklyPreviousOccurrence(TimeWindowFilterSettings settings, ZonedDateTime now) {
+        final RecurrencePattern pattern = settings.getRecurrence().getPattern();
+        final int interval = pattern.getInterval();
+        final ZonedDateTime firstDayOfFirstWeek = settings.getStart().minusDays(
+            TimeWindowUtils.daysPassedWeekStart(settings.getStart().getDayOfWeek(), pattern.getFirstDayOfWeek()));
+
+        final long numberOfInterval = Duration.between(firstDayOfFirstWeek, now).toSeconds()
+            / Duration.ofDays((long) interval * RecurrenceConstants.DAYS_PER_WEEK).toSeconds();
+        final ZonedDateTime firstDayOfMostRecentOccurrence = firstDayOfFirstWeek.plusDays(
+            numberOfInterval * (interval * RecurrenceConstants.DAYS_PER_WEEK));
+        final List<DayOfWeek> sortedDaysOfWeek = TimeWindowUtils.sortDaysOfWeek(pattern.getDaysOfWeek(), pattern.getFirstDayOfWeek());
+        final int maxDayOffset = TimeWindowUtils.daysPassedWeekStart(sortedDaysOfWeek.get(sortedDaysOfWeek.size() - 1), pattern.getFirstDayOfWeek());
+        final int minDayOffset = TimeWindowUtils.daysPassedWeekStart(sortedDaysOfWeek.get(0), pattern.getFirstDayOfWeek());
+        ZonedDateTime mostRecentOccurrence = null;
+        int numberOfOccurrences = (int) (numberOfInterval * sortedDaysOfWeek.size()
+            - (sortedDaysOfWeek.indexOf(settings.getStart().getDayOfWeek())));
+
+        // now is after the first week
```

Comment:
> // now is after the first week

this comment is not correct

Suggest: "now is not within the most recent occurring week"

Reply:
Updated